Refactor masstranslate recipe dependencies.

jone opened the pull request:
By making the "oauth2client" dependency an extras, we are not always installing "six". This reduces versions conflicts of "six" between the "oauth2client" package and Plone.

The recipe does automatically include the masstranslate-extras when the recipe is installed.

Close #8

lukasgraf commented on setup.py:
```diff
@@ -4,12 +4,22 @@
+        # oauth2client 1.4.12 is the first version with acceptable dependency
+        # declaration. So we take at least this one. NOTE: This may breake
+        # compatibility with Plone 4.3.2 and older.
+        'oauth2client <= 1.4.12',
```
The comment says "at least" but the constraint is "less than" - which one is right?

lukasgraf commented:
Could you also update the **Problems with six** section in the `README`?

jone commented:
Updates:
- Also moved the dependencies `keyring` and `gspread` into the `masstranslate` extras.
- Removed the version constraint from `oauth2client`: we've added this initially for being automatically compatible with the Plone KGS. Since we now only install it for masstranslate and I expect masstranslate buildouts to not inherit the Plone KGS, this is no longer a problem and we can remove the constraint. Users now can use version pinnings if they have conflicts with other parts of their buildouts.
- Removed the readme section, since I believe everything is great now :stuck_out_tongue_winking_eye:

lukasgraf commented:
:+1:
